Code/Framework/AzFramework/CMakeLists.txt

```diff
@@ -33,6 +32,14 @@ ly_add_target(
             3rdParty::lz4
 )
 
+set(LY_SEARCH_MODE_DEFINE $<$<NOT:$<STREQUAL:"${LY_ARCHIVE_FILE_SEARCH_MODE}","">>:LY_ARCHIVE_FILE_SEARCH_MODE_DEFAULT=${LY_ARCHIVE_FILE_SEARCH_MODE}>)
```

Comment:
a slightly simpler way to write it is:
```
set(LY_SEARCH_MODE_DEFINE "$<$<BOOL:${LY_ARCHIVE_FILE_SEARCH_MODE}>:LY_ARCHIVE_FILE_SEARCH_MODE_DEFAULT=${LY_ARCHIVE_FILE_SEARCH_MODE}>"
```
(since empty strings evaluate to false)

Reply:
I didn't use the `$<BOOL:>` generator expression because the value of `0` also evaluates to false

If the user explicitly sets `0` the define should be added as `LY_ARCHIVE_FILE_SEARCH_MODE_DEFAULT=0`